redash/serializers/__init__.py

```diff
@@ -249,20 +250,22 @@ def serialize_job(job):
         updated_at = 0
 
     status = STATUSES[job_status]
+    error_origin = ''
+    error = ''
 
     if isinstance(job.result, Exception):
+        error_origin = 'Data source' if isinstance(job.result, QueryExecutionError) else 'System'
```

Comment:
We use `QueryExecution` error in the catch all handler as well, so assuming that anything with `QueryExecutionError` is non system is probably very optimistic. Also some errors that we do handle in the query runners might be on our end. I think this is logic that should be implemented at the Query Runner level and each Query Runner will qualify the error messages.